lib/rouge/lexers/yaml.rb

```diff
@@ -328,7 +328,7 @@ def set_indent(match, opts={})
       state :plain_scalar_in_block_context do
         # the : indicator ends a scalar
         rule %r/[ ]*(?=:[ \n]|:$)/, Text, :pop!
-        rule %r/[ ]*:/, Str
+        rule %r/[ ]*:\S+/, Str
```

Comment:
I don't follow the significance of `\S+` here. Is this related to the number issue as well?

Reply:
Oh looks like I forgot to note this in the opening comment..
The deal here is to handle the datetime strings such as `time: 20:03:20`. Without that `\S`, the match would end just after the colon. But we want the match to proceed till the scanner encounters a whitespace character.

Reply:
Ahhh... make sense.

Reply:
Oops, I did have an additional follow-up.

When I check the display of YAML on the current master in the visual testing app, I don't see any problems in the highlighting of the various time values that are in the sample. Was there a specific form that was causing an issue?

Reply:
Once you add just `rule %r/\d+(?:\.\d+)?(?=(\r?\n)| +#)/, Literal::Number, :pop!` to `ln#342` on `master`, you'll notice the discrepancy with the time strings.

Reply:
Now it's all clear. Thanks!